crates/uv-distribution/src/metadata/lowering.rs

```diff
@@ -242,10 +243,21 @@ fn path_source(
     let ascend_to_workspace = project_dir
         .strip_prefix(workspace_root)
         .expect("Project must be below workspace root");
-    Ok(RequirementSource::Path {
-        install_path: path_buf,
-        lock_path: ascend_to_workspace.join(project_dir),
-        url,
-        editable,
-    })
+    if path_buf.is_dir() {
```

Comment:
Not really a huge fan of _where_ we're detecting this but I don't know that we have much of a choice.

Reply:
Is there any other safe guard / enforcement?